src/libmongoc/src/mongoc/mongoc-cluster-private.h

```diff
@@ -105,6 +105,16 @@ mongoc_cluster_try_recv (mongoc_cluster_t *cluster,
                          mongoc_server_stream_t *server_stream,
                          bson_error_t *error);
 
+/**
+ * @brief Flag telling whether an aggregate operation has a write stage or not
+ */
+typedef enum aggr_with_write_stage_flag {
+   /** The operation is not an aggregate with a write stage */
+   NOT_AGGR_WITH_WRITE_STAGE,
```

Comment:
I assume the enum starts at 0 but should we explicitly define that? A number of other enums in libmongoc also define the first value, but it's possible that's only done when a non-zero value is used.

A separate observation was that it seemed a bit odd to introduce a mandatory parameter referring to aggregation on a common method used by things other than aggregation. I'm not too concerned either way as `mongoc_cluster_stream_for_reads` is private so we could always address this down the line if desired/possible. This is really a complaint about how annoying it is to implement this feature as opposed to a criticism on the PR itself.

Edit: this may relate to the other discussion in https://github.com/mongodb/mongo-c-driver/pull/904/files#r780560871